src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -2235,61 +2233,92 @@ public Type visitErrorType(ErrorType t, Symbol sym) {
             }
         };
 
-    /**
-     * Return the base type of t or any of its outer types that starts
-     * with the given symbol.  If none exists, return null.
-     *
-     * @param t a type
-     * @param sym a symbol
-     */
+    /// Traverses a sequence of types starting with `t` and returns the first type
+    /// that can be seen as a supertype of one of those types.
+    ///
+    /// The sequence of types starts with `t` and the next type in the sequence
+    /// is obtained by calling `getEnclosingType()` on  the previous type in the
+    /// sequence.
+    ///
+    /// Example: The field `f` needs to be resolved. Its site is `Outer.Sub.I` and
+    /// outer type is its owner `Sup`. `asOuterSuper` will use the enclosing types
+    /// (examines only enclosing instances) to discover for which type `Sup` can be
+    /// seen as super of in the sequence `Outer.Sub.I`.
+    ///
+    /// Can `Sup` be seen as super of  `Outer.Sub.I`? No. Going to the next enclosing type .
+    /// Can `Sup` be seen as super of `Outer.Sub`? Yes! It’s outer results in `Sup<String>`.
+    ///
+    /// ```
+    /// static class Sup<F> { public F f; }
+    ///  class Outer {
+    ///   static class Sub extends Sup<String> {
+    ///       class I {
+    ///         void test() {
+    ///             String f2 = f; // Sup<String>
+    ///         }
+    ///       }
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// @implNote this is typically used to compute the implicit qualifier in a
```

Comment:
```suggestion
    /// @implNote This is typically used to compute the implicit qualifier in a
```
This is a new paragraph in a block tag